src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/facet/FacetLoaderService.java

```diff
@@ -0,0 +1,429 @@
+                        child,
+                        categoryName,
+                        accum,
+                        nextCollector,
+                        facetCleared,
+                        effectiveFlat,
+                        effectiveGroups // pass down groups as scope as well for completeness (not used directly)
+                );
+                created += c.created();
+                updated += c.updated();
+            }
+
+            // Parent should only have the union of its children’s mapped concepts.
+            // Always clear existing parent mappings before rebuilding from children
+            facetConceptRepository.deleteAllForFacetIds(java.util.List.of(facet.getFacetId()));
+
+            // Insert union of direct children’s concept mappings into the parent
+            mapped = facetConceptRepository.mapParentToUnionOfDirectChildren(facet.getFacetId());
+        }
+
+        accum.facetMappings().add(new FacetMappingBreakdown(categoryName, name, mapped));
+        return new Counts(created, updated);
+    }
+
+    /** Legacy mapping by flat AND list. Implementation assumed to exist previously. */
+    private long mapFacetToConcepts(Long facetId, List<FacetExpressionDTO> effective) {
+        // Existing implementation assumed; if not present, implement a simple version here.
+        // For safety, implement a streaming mapper identical in semantics to legacy evaluator.
+
+        facetConceptRepository.deleteAllForFacetIds(java.util.List.of(facetId)); // ensure clean state before remap
+
+        long inserted = 0;
+        try (java.util.stream.Stream<ConceptPathRow> rows = conceptRepository.streamLeafNodeIdAndPath()) {
+            List<Long> batch = new ArrayList<>(1024);
+            for (Iterator<ConceptPathRow> it = rows.iterator(); it.hasNext(); ) {
+                ConceptPathRow row = it.next();
+                if (FacetExpressionEvaluator.facetAppliesToConceptPath(effective, row.getConceptPath())) {
+                    batch.add(row.getConceptNodeId());
+                    if (batch.size() >= 1000) {
+                        inserted += facetConceptRepository.bulkMap(facetId, batch);
+                        batch.clear();
+                    }
+                }
+            }
+            if (!batch.isEmpty()) {
+                inserted += facetConceptRepository.bulkMap(facetId, batch);
+            }
+        }
+        return inserted;
+    }
+
+    /** New mapping for OR-of-AND groups. */
+    private long mapFacetToConceptsGrouped(Long facetId, List<List<FacetExpressionDTO>> groups) {
+        // Strategy:
+        // - Clear existing
+        // - For each concept, accept if ANY group matches (OR)
+        // - Bulk insert mappings
+        facetConceptRepository.deleteAllForFacetIds(java.util.List.of(facetId)); // ensure clean state before remap
+
+        long inserted = 0;
+        try (java.util.stream.Stream<ConceptPathRow> rows = conceptRepository.streamLeafNodeIdAndPath()) {
```

Comment:
idk what you're doing that makes these fully qualified class names, but please don't do it any more